src/components/textInput/TextInputBase.tsx

```diff
@@ -0,0 +1,348 @@
+  const [isSecretInput, setSecretInput] = React.useState<boolean>(
+    secretInput ?? false
+  );
+
+  const inputRef = useRef<TextInput>(null);
+
+  const boxStyle: ViewStyle = useMemo(() => {
+    if (inputStatus === "focused") {
+      return {
+        borderColor: IOColors["blueIO-500"],
+        borderWidth: 2
+      };
+    }
+    if (inputStatus === "error") {
+      return {
+        borderColor: IOColors["error-600"],
+        borderWidth: 1
+      };
+    }
+    return {
+      borderColor: IOColors["grey-200"],
+      borderWidth: 1
+    };
+  }, [inputStatus]);
+
+  const animatedLabelProps = useAnimatedStyle(() => ({
+    fontSize: withTiming(interpolate(labelSharedValue.value, [0, 1], [16, 12])),
+    top: withTiming(interpolate(labelSharedValue.value, [0, 1], [0, -16]))
+  }));
+
+  useEffect(() => {
+    if (value.length > 0) {
+      labelSharedValue.value = 1;
+    } else {
+      if (inputStatus !== "focused") {
+        labelSharedValue.value = 0;
+      }
+    }
+  }, [labelSharedValue, value, inputStatus]);
+
+  const onTextInputPress = () => {
+    if (disabled) {
+      return;
+    }
+    labelSharedValue.value = 1;
+    setInputStatus("focused");
+    inputRef?.current?.focus();
+  };
+
+  const onChangeTextHandler = useCallback(
+    (text: string) => {
+      if (counterLimit && text.length > counterLimit) {
+        return;
+      }
+      onChangeText(text);
+    },
+    [counterLimit, onChangeText]
+  );
+
+  const onBlurHandler = useCallback(() => {
+    if (!value) {
+      labelSharedValue.value = 0;
+    }
+    if (secretInput) {
+      setSecretInput(true);
+    }
+    if (onValidate) {
+      const isValid = onValidate(value);
+      setIsValid(isValid);
+      if (!isValid) {
+        setInputStatus("error");
+        triggerHaptic("notificationError");
+        return;
+      } else {
+        triggerHaptic("notificationSuccess");
+      }
+    }
+    setInputStatus("initial");
+  }, [onValidate, value, labelSharedValue, secretInput]);
+
+  return (
+    <>
+      <Pressable
+        onPress={onTextInputPress}
+        style={[
+          inputStatus === "disabled" ? { opacity: 0.5 } : {},
+          boxStyle,
+          styles.textInput
+        ]}
+      >
+        {icon && (
+          <>
+            <Icon name={icon} size={24} />
+            <HSpacer size={8} />
+          </>
+        )}
+        <TextInput
+          editable={!disabled}
+          secureTextEntry={isSecretInput}
+          disableFullscreenUI={true}
+          blurOnSubmit={true}
+          ref={inputRef}
+          onFocus={() => {
+            setInputStatus("focused");
+            setIsValid(undefined);
+            labelSharedValue.value = 1;
+          }}
+          maxLength={counterLimit}
+          onBlur={onBlurHandler}
+          value={value}
+          onChangeText={onChangeTextHandler}
+          style={[
+            onValidate || secretInput ? { width: "80%" } : {},
```

Comment:
Setting an arbitrary value (or magic number, like `80%`) for the width will result in this kind of behavior:
![Simulator Screen Shot - iPhone 14 Pro - 2023-08-22 at 16 02 09](https://github.com/pagopa/io-app-design-system/assets/1255491/d330ec30-5a9c-4cb8-9c29-082f29aef37d)


To avoid the text overlapping the icon, we should build the style logic with flex and let it regulate the width depending on the context (with or without icon, etc...)